File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    //查询所有品牌
    public List<TbBrand> findAll();
    //添加
    void addBrand(TbBrand brand);
    //通过id查找
    TbBrand findOne(Long id);
    //修改品牌
    void updateBrand(TbBrand brand);
    //删除品牌
    void delete(Long[] ids);
    //条件查询+分页实现
    PageResult findPageByNameAndChar(TbBrand brand, int pageNum, int pageSize);
    //模板关联查询  品牌表的所有数据
    List<Map> selectBrandList();
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    //查询所有
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }
    //条件查询+分页

    this.search=function (searchEntity,pageNum,pageSize) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }

    //根据品牌查询
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    //删除品牌
    this.delete=function (selectIds) {
        return $http.get("../brand/delete.do?ids="+selectIds)
    }

    //添加
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    //修改
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    //查询模板关联   品牌下拉列表
    this.selectBrandList=function () {
        return $http.get("../brand/selectBrandList.do");
    }

})